source/initial_temperature/continental_geotherm.cc

```diff
@@ -0,0 +1,257 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/initial_temperature/continental_geotherm.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/spherical_shell.h>
+#include <aspect/boundary_temperature/interface.h>
+#include <aspect/material_model/visco_plastic.h>
+#include <aspect/heating_model/interface.h>
+
+#include <cmath>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    ContinentalGeotherm<dim>::ContinentalGeotherm ()
+    {}
+
+
+    template <int dim>
+    void
+    ContinentalGeotherm<dim>::
+    initialize ()
+    {
+      // Check that the required radioactive heating model ("compositional heating") is used
+      const std::vector<std::string> &heating_models = this->get_heating_model_manager().get_active_heating_model_names();
+      AssertThrow(std::find(heating_models.begin(), heating_models.end(), "compositional heating") != heating_models.end(),
+                  ExcMessage("The continental geotherm initial temperature plugin requires the compositional heating plugin."));
+
+      // Check that the required material model ("visco plastic") is used
+      AssertThrow((dynamic_cast<MaterialModel::ViscoPlastic<dim> *> (const_cast<MaterialModel::Interface<dim> *>(&this->get_material_model()))) != 0,
+                  ExcMessage("The continental geotherm initial temperature plugin requires the viscoplastic material model plugin."));
+    }
+
+
+    template <int dim>
+    double
+    ContinentalGeotherm<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      const double depth = this->get_geometry_model().depth(position);
+
+      return temperature(depth);
+    }
+
+
+    template <int dim>
+    double
+    ContinentalGeotherm<dim>::
+    temperature (const double depth) const
+    {
+      // Compute some constants to calculate the temperatures T1 and T2 at the interfaces
+      // between the layers upper crust/lower crust and lower crust/lithospheric mantle.
```

Comment:
Are these specific constants referred to in Turcotte and Schubert? If so, would it be possible to reference the equations?

Reply:
Same for the equations below. I find these a bit harder to read/interpret than the Chapman equations, so it would be nice to have some specific references (if possible).